numpy/_core/src/multiarray/stringdtype/dtype.c

```diff
@@ -584,6 +584,20 @@ stringdtype_get_clear_loop(void *NPY_UNUSED(traverse_context),
     return 0;
 }
 
+static int
+stringdtype_get_fill_zero_loop(void *NPY_UNUSED(traverse_context),
+                               PyArray_Descr *NPY_UNUSED(descr),
+                               int NPY_UNUSED(aligned),
+                               npy_intp NPY_UNUSED(fixed_stride),
+                               PyArrayMethod_TraverseLoop **out_loop,
+                               NpyAuxData **NPY_UNUSED(out_auxdata),
+                               NPY_ARRAYMETHOD_FLAGS *flags)
+{
+    *flags = NPY_METH_NO_FLOATINGPOINT_ERRORS;
+    *out_loop = &stringdtype_clear_loop;
+    return 0;
+}
```

Comment:
You don't need this, nor is it correct.  What you need is what I mentioned above:  Gracefully keep the promise that we will use `memset` as the zeroing logic when there is no custom implementation.